src/extension.ts

```diff
@@ -573,10 +590,10 @@ export class Extension implements RunHooks {
     // Run either locations or test ids to always be compatible with the test server (it can run either or).
     if (files.length) {
       const fileItems = files.map(f => this._testTree.testItemForFile(f)).filter(Boolean) as vscodeTypes.TestItem[];
-      await this._queueTestRun(fileItems, 'watch');
+      await this._queueWatchRun(fileItems, 'files');
```

Comment:
Hmm, didn't think about that. There can be both `files` and `testItems` present, yes, if you're watching some files and some items.
The new logic will split them apart, that can get ugly. Tests should be executing roughly in the order of the files being saved, anything else would feel broken.

I don't see how splitting files and items up within the batch would help. Executing more than one test run at a time would be pretty jumpy in the UI I assume.

What's blocking us from sending both `files` and `testItems` within the same test run? My understanding is that today, we filter them with `AND`, but maybe we could allow switching that to `OR`, just for the test server?

Reply:
There we go: https://github.com/microsoft/playwright-vscode/pull/553

Reply:
we discussed this in a VC. decided to not ship https://github.com/microsoft/playwright-vscode/pull/553 and keep this change as is. Dima's proposed change would mean that there's one command instead of two, but it wouldn't change anything about the batching